addons/mail/models/mail_thread.py

```diff
@@ -1286,6 +1286,14 @@ def _message_parse_extract_payload(self, message, save_original=False):
                     continue  # skip container
 
                 filename = part.get_filename()  # I may not properly handle all charsets
+                if part.get_content_type() == 'text/xml'\
+                        and not part.get_param('charset')\
+                        and part._payload[:4] == base64.b64encode(b"\xef\xbb\xbf").decode():
+                    # for text/xml with omitted charset, the charset is assumed to be ASCII. If the content starts with
+                    # the UTF-8 byte order mark (BOM), strip the BOM and mark the part as UTF-8. Converting it directly
+                    # to ASCII would result in invalid characters, leading to a crash.
+                    part._payload = part._payload[4:]  # get rid of the BOM (3 bytes = 4 char in base64)
+                    part.set_charset('utf-8')
```

Comment:
```suggestion
                    part.set_charset('utf-8-sig')
```
thought of a better solution